scipy/stats/_morestats.py

```diff
@@ -991,7 +987,7 @@ def boxcox_llf(lmb, data, *, axis=0, keepdims=False, nan_policy='propagate'):
                           result_to_tuple=lambda x: (x,))
 def _boxcox_llf(data, axis=0, *, lmb):
     xp = array_namespace(data)
-    data = xp_promote(data, force_floating=True, xp=xp)
+    lmb, data = xp_promote(lmb, data, force_floating=True, xp=xp)
```

Comment:
Typically, `lmb` will be a Python scalar, but this will help with NumPy scalars and 0-d arrays.